### PR TITLE
Feat/earn info modals

### PR DESCRIPTION
**Pull request by mitchellv-ledger**
### 📝 Description

Adds earn info modal drawer that is set via deeplink.
For example, `xcrun simctl openurl booted "ledgerlive://earn?action=info-modal&message=hello%20what%20are%20you&messageTitle=happy"` will open the modal in simulator with title and message as set by url params.

### ❓ Context

- **Impacted projects**: `` 
- **Linked resource(s)**: `LIVE-8855` 

### ✅ Checklist

- [] **Test coverage** 
- [X] **Atomic delivery** 
- [X] **No breaking changes** 

### 📸 Demo
Screenshot 2023-08-11 at 14 01 41

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx [hunk 1]
```diff
@@ -21,9 +23,12 @@ const Earn = (_props: StackNavigatorProps<EarnLiveAppNavigatorParamList, ScreenN
   // TODO: update to use specific mobile feature flag
```
Comment (beths-ledger) [inline apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx:23]: if you go back to these files it'd be nice to remove the no longer relevant comment on line 23 btw (just an aside / nice to have)

File: apps/ledger-live-mobile/src/components/RootDrawer/RootDrawer.tsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { useNavigation } from "@react-navigation/native";
 import { StackNavigationProp } from "@react-navigation/stack";
 import { NavigatorName } from "../../const";
 import { PTXServicesAppleWarning } from "./InitialDrawers/PTXServicesAppleWarning";
+import { EarnInfoDrawer } from "../../screens/PTX/Earn/EarnInfoDrawer";
```
Comment (beths-ledger) [inline apps/ledger-live-mobile/src/components/RootDrawer/RootDrawer.tsx:12]: Should this (and the id const `EarnInfoDrawerID`) be used in the RootDrawerSelector fn (line 33)? Or if not, can we remove the import?

File: apps/ledger-live-mobile/src/screens/PTX/Earn/EarnInfoDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import useFeature from "@ledgerhq/live-common/featureFlags/useFeature";
```
Comment (beths-ledger) [inline apps/ledger-live-mobile/src/screens/PTX/Earn/EarnInfoDrawer.tsx:2]:
```suggestion
```

**Review by beths-ledger — COMMENTED** [review]
I still need to test this manually, but I noticed a few little things you might want to look at already. Otherwise looking  👌

File: apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx (hunk 1 above)
Comment (mitchellv-ledger) [reply apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx:23]: done 👍

File: apps/ledger-live-mobile/src/screens/PTX/Earn/EarnInfoDrawer.tsx (hunk 3 above)
Comment (mitchellv-ledger) [reply apps/ledger-live-mobile/src/screens/PTX/Earn/EarnInfoDrawer.tsx]: removed

**Review by beths-ledger — APPROVED** [review]
works well

**Review by beths-ledger — APPROVED** [review]
Even better 👍

File: apps/ledger-live-mobile/src/reducers/earn.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import { EarnPayload, EarnSetInfoModalPayload } from "../actions/types";
+
+export const INITIAL_STATE: EarnState = {
+  infoModal: {},
+};
+
+const handlers: ReducerMap<EarnState, EarnPayload> = {
+  EARN_INFO_MODAL: (state, action) => ({
+    ...state,
+    infoModal: (action as Action<EarnSetInfoModalPayload>)?.payload || {},
```
Comment (beths-ledger) [inline apps/ledger-live-mobile/src/reducers/earn.ts:14]: Does the `payload` object need to be spread as well?

**Pull request merged**